src/AppBundle/Repository/NoticeRepository.php

```diff
@@ -3,66 +3,74 @@
 namespace AppBundle\Repository;
 
 use AppBundle\Entity\Notice;
+use AppBundle\Helper\NoticeVisibility;
 use Doctrine\ORM\QueryBuilder;
 
 class NoticeRepository extends BaseRepository
 {
+    public function getAll()
+    {
+        return $this->createQueryForPublicNotices()
+            ->getQuery()
+            ->getResult();
+    }
+
     /**
      * @param int|null $id
      *
      * @return Notice|null
      *
-     * @throws \Doctrine\ORM\NonUniqueResultException
+     * @throws NonUniqueResultExceptionAlias
      */
     public function getOne($id)
     {
-        $queryBuilder = $this->repository->createQueryBuilder('n')
-            ->select('n,c')
-            ->leftJoin('n.contributor', 'c')
-            ->where('n.id = :id')
-            ->andWhere('c.enabled = true')
-            ->setParameter('id', $id);
-
-        return self::addNoticeExpirationLogic($queryBuilder)->getQuery()->getOneOrNullResult();
-    }
-
-    public function getAll()
-    {
-        return $this->createQueryForPublicNotices('n')->getQuery()->getResult();
+        return $this->createQueryForPublicNotices()
+            ->andWhere('n.id = :id')
```

Comment:
Previously we were defining the alias as an option of `createQueryForPublicNotices` thus we knew where the  `n` was coming from.

Reply:
I though it was only adding noise, and that’s a convention that we use `n` for `notice` lead to simpler code … I hesitated to use `notice` instead of `n` though 

Disagree ?

Reply:
And even if we let use some other alias here, then we loose some reusability if we don’t use the same everywhere (think of `addNoticeVisibilityLogic`) for instance

Reply:
And even if we let use some other alias here, then we loose some reusability if we don’t use the same everywhere (think of addNoticeVisibilityLogic) for instance

Reply:
True, but that's why we should keep the alias as an option in all our `createQuerybuilder*`.
Everything is explicit and there's no conventions to guess. 
Not a blocker for me, but I know you are a lover of *explicit content* :wink:

Reply:
How do you find this version @lutangar  ?